### PR TITLE
errorprone static analysis

### PR DESCRIPTION
**Pull request by danielnorberg**
http://errorprone.info/

### CODE REVIEW

**Comment by danielnorberg** [comment]
@honnix PTAL

**Comment by danielnorberg** [comment]
Not going to go through the effort to get line coverage on those private constructors

**Comment by honnix** [comment]
So we give up codacy? It’s too slow because we don’t pay. :(

**Comment by danielnorberg** [comment]
I was thinking we'd use both =)

**Comment by danielnorberg** [comment]
errorprone is useful because it's part of our local build cycle. Catch issues before pushing and avoid  delayed codacy failures...

File: styx-api-service/src/test/java/com/spotify/styx/api/ApiVersionTestUtils.java [hunk 1]
```diff
@@ -29,6 +29,10 @@
  */
 class ApiVersionTestUtils {
 
+  private ApiVersionTestUtils() {
+    throw new UnsupportedOperationException();
```
Comment (honnix) [inline styx-api-service/src/test/java/com/spotify/styx/api/ApiVersionTestUtils.java]: Is this the new best practice? Like making it really impossible to construct an instance. I wish Java had `object`.

Comment (danielnorberg) [reply styx-api-service/src/test/java/com/spotify/styx/api/ApiVersionTestUtils.java]: Yeah, from Effective Java: http://www.informit.com/articles/article.aspx?p=1216151&seqNum=4

Comment (honnix) [reply styx-api-service/src/test/java/com/spotify/styx/api/ApiVersionTestUtils.java]: Ok. I could only recall the private constructor part not the exception. :)

**Pull request merged**